grove/tomography/tomography.py

```diff
@@ -0,0 +1,727 @@
+        constraints = [p_jkl >= 0,
+                       np.matrix(np.ones((1, pi_basis.dim))) * p_jkl_mat == 1]
+        r_mn_mat = cvxpy.reshape(r_mn, pauli_basis.dim, pauli_basis.dim)
+        super_pauli_basis = pauli_basis.super_basis()
+        choi_real_imag = sum((r_mn_mat[jj, kk] * ut.to_realimag(
+            super_pauli_basis.ops[jj + kk * pauli_basis.dim])
+                              for jj in range(pauli_basis.dim)
+                              for kk in range(pauli_basis.dim)), 0)
+
+        if COMPLETELY_POSITIVE in settings.constraints:
+            if _SDP_SOLVER.is_functional():
+                constraints.append(choi_real_imag >> 0)
+            else:  # pragma no coverage
+                _log.warning("No convex solver capable of semi-definite problems installed.\n"
+                             "Dropping the complete positivity constraint on the process")
+
+        if TRACE_PRESERVING in settings.constraints:
+            constraints.append(r_mn_mat[0, 0] == 1)
+            constraints.append(r_mn_mat[0, 1:] == 0)
+
+        prob = cvxpy.Problem(cvxpy.Minimize(obj), constraints)
+        _ = prob.solve(solver=SOLVER, **settings.solver_kwargs)
+
+        r_mn_est = r_mn.value.reshape((pauli_basis.dim, pauli_basis.dim)).transpose()
+        return ProcessTomography(r_mn_est, pauli_basis, settings)
+
+    def __init__(self, r_est, pauli_basis, settings):
+        """
+        Construct a ProcessTomography to encapsulate the result of estimating a quantum process
+        from a quantum tomography measurement.
+
+        :param numpy.ndarray r_est: The estimated quantum process represented as a Pauli transfer
+         matrix.
+        :param OperatorBasis pauli_basis: The employed (generalized) Pauli basis.
+        :param TomographySettings settings: The settings used to estimate the process.
+        """
+        self.r_est = r_est
+        self.sop = pauli_basis.super_from_tm(r_est)
+        self.pauli_basis = pauli_basis
+        self.settings = settings
+
+    def process_fidelity(self, other):
+        """
+        Compute the quantum process fidelity of the estimated state with respect to a unitary
+        process.
+
+        :param (qutip.Qobj|matrix-like) other: A unitary operator that induces a process as
+         ``rho -> other*rho*other.dag()``, can also be a superoperator or Pauli-transfer matrix.
+        :return: The process fidelity, a real number between 0 and 1.
+        :rtype: float
+        """
+        if isinstance(other, qt.Qobj):
+            if not other.issuper or other.superrep != SUPER:
+                sother = qt.to_super(other)
+            else:
+                sother = other
+            tm_other = self.pauli_basis.transfer_matrix(sother)
+        else:
+            tm_other = csr_matrix(other)
+        dimension = self.pauli_basis.ops[0].shape[0]
+        return np.trace(tm_other.T * self.r_est).real / dimension ** 2
+
+    def avg_gate_fidelity(self, other):
```

Comment:
this signature is misleading to me. With `other` I associate an object of the same type as this class, i.e. a Tomogrpahy instance

Reply:
:+1: